crates/rust-analyzer/src/lsp_ext.rs

```diff
@@ -221,6 +221,9 @@ pub struct SsrParams {
     /// position.
     #[serde(flatten)]
     pub position: lsp_types::TextDocumentPositionParams,
+
+    /// Current selections. Search/replace will be restricted to these if non-empty.
+    pub selections: Vec<lsp_types::Range>,
```

Comment:
I think long-term we want something smarter here, like the list of files with selections globs or whatnot, but this seems like a fine start .